File: Frontend/types/models/Feedback.ts
export enum FeedbackRole {
  STUDENT = "student",
  TEACHER = "teacher",
}

export enum Category {
  COMPLAINT = "complaint",
  SUGGESTION = "suggestion",
  APPRECIATION = "appreciation",
  OTHER = "other",
}

export enum FeedbackStatus {
  UNREAD = "unread",
  REVIEWING = "reviewing",
  RESOLVED = "resolved",
}

export interface FeedbackResponse {
  responder_id?: string;
  message?: string;
  responded_at?: string;
}

export interface Feedback {
  _id?: string;
  sender_id: string;
  receiver_id?: string | null;
  role: FeedbackRole;
  category: Category;
  message: string;
  response?: FeedbackResponse | null;
  status: FeedbackStatus;
  created_at?: string;
}

File: Backend/app/models/feedback.py
from pydantic import BaseModel, Field # type: ignore
from datetime import datetime, timezone
from app.enums.roles import FeedbackRole
from app.enums.category import Category
from app.enums.status import FeedbackStatus
from app.utils.pyobjectid import PyObjectId
from typing import ClassVar

class FeedbackResponseModel(BaseModel):
    _collection_name: ClassVar[str] = "feedback_response"
    id: PyObjectId | None = Field(None, alias="_id")
    responder_id: str | None = None
    message: str | None = None
    responded_at: datetime | None = None


class FeedbackModel(BaseModel):
    id: PyObjectId | None = Field(default=None, alias="_id")
    sender_id: str
    receiver_id: str | None = None
    role: FeedbackRole
    category: Category
    message: str = Field(..., min_length=5, max_length=1000)
    status: FeedbackStatus | None = None
    response: FeedbackResponseModel | None = None
    created_at: datetime | None = None
    model_config = {
        "extra": "ignore",
        "from_attributes": True,
        "populate_by_name": True,
        "use_enum_values": True,
        "arbitrary_types_allowed": True, 
        "json_encoders": {

            PyObjectId: str,
            datetime: lambda dt: dt.isoformat(),
        }
    }
